src/Storages/Kafka/KafkaBlockInputStream.cpp

```diff
@@ -186,17 +193,17 @@ Block KafkaBlockInputStream::readImpl()
 
             total_rows = total_rows + new_rows;
         }
-        else if (buffer->isStalled())
-        {
-            ++failed_poll_attempts;
-        }
         else if (buffer->polledDataUnusable())
         {
             break;
         }
+        else if (buffer->isStalled())
+        {
+            ++failed_poll_attempts;
+        }
         else
         {
-            LOG_WARNING(log, "Parsing of message (topic: {}, partition: {}, offset: {}) return no rows.", buffer->currentTopic(), buffer->currentPartition(), buffer->currentOffset());
```

Comment:
what happened with that line?

Reply:
Because it may also bail, i.e. what will happen if buffer will be in REBALANCE_HAPPENED state?

Reply:
Then if branch above that line will catch it.

Reply:
If think you're afraid to get that state because of rebalance callback. But it is synchronious and called only during poll

Reply:
>Then if branch above that line will catch it.

This states are too tricky, but you are right